src/Operation/Chunk.php

```diff
@@ -18,47 +18,49 @@
 final class Chunk extends AbstractOperation
 {
     /**
-     * @return Closure(int...): Closure(iterable<TKey, T>): Generator<int, list<T>>
+     * @return Closure(int...): Closure(iterable<TKey, T>): Generator<int, non-empty-list<T>>
      */
     public function __invoke(): Closure
     {
         return
             /**
-             * @return Closure(iterable<TKey, T>): Generator<int, list<T>>
+             * @return Closure(iterable<TKey, T>): Generator<int, non-empty-list<T>>
              */
             static fn (int ...$sizes): Closure =>
                 /**
                  * @param iterable<TKey, T> $iterable
                  *
-                 * @return Generator<int, list<T>>
+                 * @return Generator<int, non-empty-list<T>>
                  */
                 static function (iterable $iterable) use ($sizes): Generator {
                     $sizesCount = count($sizes);
-                    $i = 0;
-                    $values = [];
+                    $chunkIndex = 0;
+                    $chunk = [];
 
                     foreach ($iterable as $value) {
-                        $size = $sizes[$i % $sizesCount];
+                        $size = $sizes[$chunkIndex % $sizesCount];
 
                         if (0 >= $size) {
                             return;
                         }
 
-                        if (count($values) !== $size) {
-                            $values[] = $value;
+                        if (count($chunk) !== $size) {
+                            $chunk[] = $value;
 
                             continue;
                         }
 
-                        ++$i;
+                        ++$chunkIndex;
 
-                        yield $values;
+                        /** @var non-empty-list<T> $chunk */
 
```

Comment:
without this `@var` annotation psalm won't be able to infer the fact that chunk is never empty when it is yielded

Reply:
Actually this didn't help either, because psalm treats `T` somewhat awkwardly.

Threrefore, I have somewhat rewritten the implementation into following variant so that both psalm and phpstan are satisfied:

```php
$chunk[] = $value;

if (count($chunk) >= $size) {
    ++$chunkIndex;

    yield $chunk;

    $chunk = [];
}
```